src/lib/datepicker/calendar.ts

```diff
@@ -200,6 +220,17 @@ export class MatCalendar<D> implements AfterContentInit, OnDestroy, OnChanges {
   }
 
   ngAfterContentInit() {
+
+    if (this.customCalendarHeaderComponent !== undefined) {
```

Comment:
I think you can just simplify this logic to:

```ts
this.clalendarHeaderPortal = new ComponentPortal(this.calendarHeaderComponent || MatCalendarHeader);
```

Reply:
done in 3467c0ac390cb4401cf7c44b79c1280ba5dcb5f8

In general I try to be as specific as possible and not rely on implicit type conversions. JavaScript may behave very unexpectedly in some cases. 

This is my understanding of the construct:
If `this.calendarHeaderComponent` is set, we expect it to be evaluated to true and thus returned. If not (`undefined`), to false and thus  `MatCalendarHeader` to be returned.